mlflow/lightgbm.py

```diff
@@ -484,10 +489,23 @@ def log_feature_importance_plot(features, importance, importance_type):
 
             # If early_stopping_rounds is present, logging metrics at the best iteration
             # as extra metrics with the max step + 1.
-            early_stopping_index = all_arg_names.index("early_stopping_rounds")
-            early_stopping = (
-                num_pos_args >= early_stopping_index + 1 or "early_stopping_rounds" in kwargs
-            )
+            if Version(lightgbm.__version__) <= Version("3.3.1"):
+                # The parameter `early_stopping_rounds` in `lightgbm.train` is removed in this PR:
+                # https://github.com/microsoft/LightGBM/pull/4908
+                early_stopping_index = all_arg_names.index("early_stopping_rounds")
+                early_stopping = (
+                    num_pos_args >= early_stopping_index + 1 or "early_stopping_rounds" in kwargs
+                )
+            else:
+                early_stopping = False
+                if "callbacks" in kwargs and kwargs["callbacks"] is not None:
+                    for cb in kwargs["callbacks"]:
+                        if (
+                            hasattr(cb, "__qualname__")
+                            and cb.__qualname__ == "early_stopping.<locals>._callback"
```

Comment:
This checking is a little tricky , is there better way ?  I think you want to check whether the callback is `lgb.early_stopping(5)` , can we directly compare the function object with it ?

Reply:
Directly comparing function object seems not working:
```python
lightgbm.early_stopping(stopping_rounds=1) is lightgbm.early_stopping(stopping_rounds=1) # False
lightgbm.early_stopping(stopping_rounds=1) == lightgbm.early_stopping(stopping_rounds=1) # False
```